python/tempo/tsdf.py

```diff
@@ -216,11 +217,19 @@ def __getLastRightRow(
 
         sort_keys = [f.col(col_name) for col_name in ptntl_sort_keys if col_name != ""]
 
-        window_spec = (
-            Window.partitionBy(self.partitionCols)
-            .orderBy(sort_keys)
-            .rowsBetween(Window.unboundedPreceding, Window.currentRow)
-        )
+        if tolerance is not None:
+            window_spec = (
+                Window.partitionBy(self.partitionCols)
+                .orderBy(f.col(self.ts_col).cast("long"))
```

Comment:
`ts_col` should convert to double, not long (there could be fractional seconds in the Timestamp)
`sort_keys` also includes other columns that are relevant in this function.

Reply:
Thanks for the update and new method.